test/unit/bypass_validation_tests.js

```diff
@@ -0,0 +1,243 @@
+'use strict';
+
+const MongoClient = require('../..').MongoClient;
+const expect = require('chai').expect;
+const mock = require('mongodb-mock-server');
+
+describe('bypass document validation', function() {
+  const test = {};
+  beforeEach(() => {
+    return mock.createServer().then(server => {
+      test.server = server;
+    });
+  });
+  afterEach(() => mock.cleanup());
+
+  // general test for aggregate function
+  function testAggregate(bypassDocumentValidation, done) {
+    const client = new MongoClient(`mongodb://${test.server.uri()}/test`);
+    let close = e => {
+      close = () => {};
+      client.close(() => done(e));
+    };
+
+    test.server.setMessageHandler(request => {
+      var doc = request.document;
+      if (doc.aggregate) {
+        try {
+          expect(doc.bypassDocumentValidation).equal(bypassDocumentValidation || undefined);
+          request.reply({
+            ok: 1,
+            cursor: {
+              firstBatch: [{}],
+              id: 23,
+              ns: 'test.test'
+            }
+          });
+        } catch (e) {
+          close(e);
+        }
+      }
+
+      if (doc.ismaster) {
+        request.reply(Object.assign({}, mock.DEFAULT_ISMASTER));
+      } else if (doc.endSessions) {
+        request.reply({ ok: 1 });
+      }
+    });
+
+    client.connect(function(err, client) {
+      expect(err).to.not.exist;
+      var db = client.db('test');
+      var collection = db.collection('test_c');
+
+      var options = {
+        bypassDocumentValidation: bypassDocumentValidation
+      };
+
+      var pipeline = [
+        {
+          $project: {}
+        }
+      ];
+      collection.aggregate(pipeline, options).next(() => close());
+    });
+  }
+  // aggregate
+  it('should only set bypass document validation if strictly true in aggregate', function(done) {
+    testAggregate(true, done);
```

Comment:
Instead of passing in a boolean, you can pass in an object:

```js
testAggregate({ expected: true, actual: true }, done);
```

This is also true for the bulkWrite tests below, where you are passing a flag for ordered.